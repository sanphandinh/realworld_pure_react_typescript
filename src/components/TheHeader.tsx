import React, { FC } from 'react';
import TheHeaderNavItem from './TheHeader.NavItem';
import { useAuthState } from '../providers/AuthProvider/hooks';
import {
  HomeRoute,
  EditorRoute,
  SettingsRoute,
  SigninRoute,
  SignupRoute,
} from '../constants/routes.constants';
import { Link } from '@reach/router';

const TheHeader: FC = () => {
  const { isLogin } = useAuthState();
  return (
    <nav className="navbar navbar-light">
      <div className="container">
        <Link className="navbar-brand" to={HomeRoute}>
          Conduit
        </Link>
        <ul className="nav navbar-nav pull-xs-right">
          <TheHeaderNavItem title="Home" url={HomeRoute} />
          {isLogin ? (
            <>
              <TheHeaderNavItem
                title={
                  <>
                    <i className="ion-compose"></i>&nbsp;New Article
                  </>
                }
                url={EditorRoute}
              />
              <TheHeaderNavItem
                title={
                  <>
                    <i className="ion-gear-a"></i>&nbsp;Settings
                  </>
                }
                url={SettingsRoute}
              />
            </>
          ) : (
            <>
              <TheHeaderNavItem title="Sign in" url={SigninRoute} />
              <TheHeaderNavItem title="Sign up" url={SignupRoute} />
            </>
          )}
        </ul>
      </div>
    </nav>
  );
};

export default TheHeader;
