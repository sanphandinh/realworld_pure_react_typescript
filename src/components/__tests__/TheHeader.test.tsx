import React from 'react';
import TheHeader from '../TheHeader';
import renderer from 'react-test-renderer';

test('The header render correctly', () => {
  const cmp = renderer.create(<TheHeader />);
  const tree = cmp.toJSON();
  expect(tree).toMatchSnapshot();
});
