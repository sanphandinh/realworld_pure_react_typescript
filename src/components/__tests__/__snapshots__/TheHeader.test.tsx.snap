// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The header render correctly 1`] = `
<nav
  className="navbar navbar-light"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="index.html"
    >
      conduit
    </a>
    <ul
      className="nav navbar-nav pull-xs-right"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link active"
          href="/"
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
        >
          <i
            className="ion-compose"
          />
           New Post
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
        >
          <i
            className="ion-gear-a"
          />
           Settings
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
        >
          Sign up
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
